ogcapi-draft/ogcapi-observation-processing/src/main/java/de/ii/ldproxy/ogcapi/observation_processing/parameters/QueryParameterCoordPosition.java

```diff
@@ -147,7 +146,7 @@ public boolean isEnabledForApi(OgcApiApiDataV2 apiData, String collectionId) {
         if (coord==null) {
             coord = getDefault(apiData, Optional.of(featureType.getId())).orElse(null);
             if (coord == null)
-                throw new BadRequestException(String.format("The required parameter '%s' has no value.", getName()));
+                throw new IllegalStateException(String.format("The required parameter '%s' has no value.", getName()));
```

Comment:
If a parameter value is missing/incorrect, this sounds like an IllegalArgumentException to me. 

IllegalStateException is "signals that a method has been invoked at an illegal or inappropriate time", which is not the reason for this issue.

I think we probably should add [exception guidance](https://github.com/interactive-instruments/ldproxy/blob/master/docs/dev/exceptions.md) how to map situations to standard Java exceptions and how these are eventually mapped to Web exceptions. Could you maybe summarize how you mapped which situations as a starting point?